File: C++/binary-search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int i = 0, j = nums.size() - 1;
        while (i <= j) {
            int k = (i + j) / 2;
            if (nums[k] == target) {
                return k;
            } else if (target < nums[k]) {
                j = k - 1;
            } else {
                i = k + 1;
            }
        }
        return -1;
    }
};

File: Kotlin/binary-search.kt
class Solution {
    fun search(nums: IntArray, target: Int): Int {
        var lo = 0
        var hi = nums.size - 1
        while (lo <= hi) {
            val mid = (lo + hi) / 2
            if (target == nums[mid]) {
                return mid
            } else if (target < nums[mid]) {
                hi = mid - 1
            } else {
                lo = mid + 1
            }
        }
        return -1
    }
}